docs-ui/stories/components/profiling/flamegraphZoomView.stories.js

```diff
@@ -15,7 +16,9 @@ export const EventedTrace = () => {
     <FlamegraphPreferencesProvider>
       <FlamegraphThemeProvider>
         <FullScreenFlamegraphContainer>
-          <Flamegraph profiles={eventedProfiles} />
+          <FlamegraphStateProvider>
```

Comment:
Will this provider support other functionalities beyond search? If not, can we give it a clearer name, perhaps `FlamegraphSearchStateProvider` to indicate what type of state it stores? The same for all the associated hooks.

Reply:
Yup 100%, I have a follow up PR that renames this, I just need to fix a bug in the reducer first